### DISCUSSION TITLE
[Question]: error: KeyvMongo connection error: connection timed out

### DISCUSSION THREAD

**Discussion by JIPINGBABA** [danny-avila/LibreChat#4963] [Troubleshooting]
### What is your question?

I want to perform secondary development on LibreChat, so I need to start it using npm.but I use -> npm run backend

info: [Optional] Redis not initialized. Note: Redis support is experimental.
2024-12-12 13:56:40 error: KeyvMongo connection error: connection timed out
2024-12-12 13:56:40 error: There was an uncaught error: connection timed out

could you kindly help me,thanks

### More Details

![image](https://github.com/user-attachments/assets/fb22073a-8535-4d0d-b5ca-d36b22859045)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JIPINGBABA** [comment]
Hope somebody can help me,thanks

**Comment by danny-avila** [comment]
Read the docs for more info on development: https://www.librechat.ai/docs/development/get_started

**Reply by JIPINGBABA** [reply]
Thanks so much.

**Comment by ptrwis** [comment]
> ### What is your question?
> I want to perform secondary development on LibreChat, so I need to start it using npm.but I use -> npm run backend
> 
> info: [Optional] Redis not initialized. Note: Redis support is experimental. 2024-12-12 13:56:40 error: KeyvMongo connection error: connection timed out 2024-12-12 13:56:40 error: There was an uncaught error: connection timed out
> 
> could you kindly help me,thanks


Try this:  in a file "keyvMongo.js":

```
const keyvMongo = new KeyvMongo(MONGO_URI, { collection: 'logs', 
	maxIdleTimeMS: 80000,
	serverSelectionTimeoutMS: 80000,
	socketTimeoutMS: 0,
	connectTimeoutMS: 0,
 });
```

**Reply by geodanchev** [reply]
Is there any reason why this is so slow on the dev container environment ?
